processor/elasticinframetricsprocessor/processor_test.go

```diff
@@ -111,6 +111,85 @@ func TestProcessMetrics(t *testing.T) {
 	}
 }
 
+func TestDropOriginalMetrics(t *testing.T) {
+	testCases := []struct {
+		name                   string
+		cfg                    *Config
+		createMetrics          func() pmetric.Metrics
+		expectedPodname        string
+		expectedMetricName     string
+		expetedMetricValue     float64
+		expetedLengthOfMetrics int
+	}{
+		{
+			name: "ProcessMetrics when AddK8sMetrics is enabled and DropOriginal is enabled",
+			cfg:  &Config{AddK8sMetrics: true, DropOriginal: true},
+			createMetrics: func() pmetric.Metrics {
+				md := pmetric.NewMetrics()
+				rm := md.ResourceMetrics().AppendEmpty()
+				rm.Resource().Attributes().PutStr("k8s.pod.name", "test-pod")
+				sm := rm.ScopeMetrics().AppendEmpty()
+				// Creating first metric
+				metric1 := sm.Metrics().AppendEmpty()
+				metric1.SetName("k8s.pod.cpu_limit_utilization")
```

Comment:
Shouldn't we have 3 types of metrics here?

- metrics that are not mapped at all and hence are dropped.
- original metrics that are mapped but are dropped.
- remapped metrics that are kept.

Reply:
I think that two first categories are the same eventually and we dont care for the `metrics that are not mapped at all and hence are dropped.`

At the moment I dont see the remapper working correctly calling the external lib. It was not working also [here](https://github.com/elastic/opentelemetry-collector-components/blob/49e16899f84cf4e490c961d2fd38c922fc773558/processor/elasticinframetricsprocessor/processor_test.go#L55)  I think.

I will keep troubleshooting

Reply:
@ChrsMark tests finally are fixed. Let me know